### PR TITLE
jetbrains.idea-community: add darwin support

### PR DESCRIPTION
**Pull request by uri-canva**
###### Motivation for this change
Add darwin support to `jetbrains.idea-community`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by uri-canva** [comment]
cc @Mic92

**Comment by uri-canva** [comment]
ping @edwtjo

**Comment by uri-canva** [comment]
ping @edwtjo

**Comment by uri-canva** [comment]
cc @sonercirit @baracoder @c0bw3b

**Review by baracoder — APPROVED** [review]
I am not an idea user, but, so I did not test in depth, but the changes look good to me.

File: pkgs/applications/editors/jetbrains/common.nix [hunk 1]
```diff
@@ -28,7 +28,7 @@ with stdenv; lib.makeOverridable mkDerivation rec {
 
   nativeBuildInputs = [ makeWrapper patchelf p7zip unzip ];
 
-  patchPhase = ''
+  patchPhase = if stdenv.hostPlatform.system == "x86_64-darwin" then "" else ''
```
Comment (Mic92) [inline pkgs/applications/editors/jetbrains/common.nix]:
```suggestion
  patchPhase = optionalString (stdenv.hostPlatform.system != "x86_64-darwin") ''
```

File: pkgs/applications/editors/jetbrains/common.nix [hunk 2]
```diff
@@ -63,7 +63,7 @@ with stdenv; lib.makeOverridable mkDerivation rec {
     item=${desktopItem}
 
     makeWrapper "$out/$name/bin/${loName}.sh" "$out/bin/${execName}" \
-      --prefix PATH : "$out/libexec/${name}:${stdenv.lib.makeBinPath [ jdk coreutils gnugrep which git ]}" \
+      --prefix PATH : "$out/libexec/${name}:${jdk}/jdk/Contents/Home/bin:${stdenv.lib.makeBinPath [ jdk coreutils gnugrep which git ]}" \
```
Comment (Mic92) [inline pkgs/applications/editors/jetbrains/common.nix]:
```suggestion
      --prefix PATH : "$out/libexec/${name}:${optionalString (!stdenv.isDarwin) "${jdk}/jdk/Contents/Home/bin"}:${stdenv.lib.makeBinPath [ jdk coreutils gnugrep which git ]}" \
```

File: pkgs/development/compilers/jetbrains-jdk/default.nix [hunk 3]
```diff
@@ -29,20 +34,20 @@ let drv = stdenv.mkDerivation rec {
     jrePath=$out/jre
   '';
 
-  postFixup = ''
+  postFixup = if stdenv.hostPlatform.system == "x86_64-darwin" then "" else ''
```
Comment (Mic92) [inline pkgs/development/compilers/jetbrains-jdk/default.nix]: see optionalString

File: pkgs/development/compilers/jetbrains-jdk/default.nix [hunk 4]
```diff
@@ -29,20 +34,20 @@ let drv = stdenv.mkDerivation rec {
...
     find $out -type f -perm -0100 \
         -exec patchelf --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
         --set-rpath "$rpath" {} \;
     find $out -name "*.so" -exec patchelf --set-rpath "$rpath" {} \;
   '';
 
-  rpath = lib.makeLibraryPath ([
+  rpath = if stdenv.hostPlatform.system == "x86_64-darwin" then "" else (lib.makeLibraryPath ([
```
Comment (Mic92) [inline pkgs/development/compilers/jetbrains-jdk/default.nix]: see optionalString

**Comment by uri-canva** [comment]
Addressed comments.

**Comment by vcunat** [comment]
`jetbrains.rider` wasn't among your aims, right?  9892d5ab2c

**Comment by Mic92** [comment]
I don't think rider was tested. But it might just work as it is on macOS.

**Comment by Mic92** [comment]
Thanks for fixing the evaluation!

**Comment by uri-canva** [comment]
No it wasn't, but I can have a quick look, if it's simple enough I'll fix that.

**Comment by uri-canva** [comment]
Yeah looks like it's an easy fix: https://github.com/NixOS/nixpkgs/pull/69349

File: pkgs/applications/editors/jetbrains/common.nix [hunk 5]
```diff
@@ -28,7 +28,7 @@ with stdenv; lib.makeOverridable mkDerivation rec {
 
   nativeBuildInputs = [ makeWrapper patchelf p7zip unzip ];
 
-  patchPhase = ''
+  patchPhase = lib.optionalString (!stdenv.isDarwin) ''
```
Comment (volth) [inline pkgs/applications/editors/jetbrains/common.nix]: there is already `with stdenv.lib` above, line 7
all these `lib.` prefixes are non needed

**Pull request merged**